docs/site/migration/models/mixins.md

```diff
@@ -6,8 +6,730 @@ sidebar: lb4_sidebar
+
+- placing the mixin logic in a file in a mixins directory
+- updating the server/model-config.json file with the mixin directory location
+- updating the model's json file to include the mixin's name and a boolean
+
+As an example, we will create a mixin that adds a **category** property to a
+model.
+
+#### Defining The Model Property Mixin category.js
+
+The developer defines a model property mixin in **common/mixins/category.js**
+which adds a required property named `category` to any model.
+
+{% include code-caption.html content="common/mixins/category.js" %}
+
+```js
+module.exports = function(Model, options) {
+  Model.defineProperty('category', {type: 'string', required: true});
+};
+```
+
+#### Updating model-config.json
+
+The **server/model-config.json** needs to contain:
+
+- the locations of all **models**
+- the location of all **mixins**
+- the entry of the model that receives the mixin content (for this example
+  `Note`)
+
+{% include code-caption.html content="server/model-config.json" %}
+
+```
+{
+  "_meta": {
+    "sources": [
+      "loopback/common/models",
+      "loopback/server/models",
+      "../common/models",
+      "./models"
+    ],
+    "mixins": [
+      "loopback/common/mixins",
+      "loopback/server/mixins",
+      "../common/mixins",
+      "./mixins"
+    ]
+  },
+
+ // ... other entries
+
+  "Note": {
+    "dataSource": "db"
+  }
+}
+
+```
+
+Please see
+[Reference mixins in model-config.js](https://loopback.io/doc/en/lb3/Defining-mixins.html#reference-mixins-in-model-configjs)
+for a short explanation of this file.
+
+#### Applying The category.js Mixin To A Model
+
+To extend the model `Note` with the **category.js** mixin, we need to add a
+**mixins** section in **common/models/note.json** to indicate which mixins
+should be applied to it.
+
+{% include code-caption.html content="common/models/note.json" %}
+
+```json
+{
+  "name": "Note",
+  "properties": {
+    "title": {
+      "type": "string",
+      "required": true
+    },
+    "content": {
+      "type": "string"
+    }
+  },
+  "mixins": {
+    "Category": true
+  }
+}
+```
+
+Specifying a value of **true** for `Category` will apply the **category.js**
+property model mixin to the `Note` model. A value of **false** will not apply
+the mixin.
+
+### LoopBack 4 Approach
+
+In LoopBack 4, a developer is able to create a model property mixin by:
+
+- creating a base model class which extends `Entity`
+- placing the mixin class factory function in a separate file
+- generating a model using the CLI as usual
+- adjusting the model file to make use of the mixin class factory function
+
+#### Defining A BaseEntity Class Which Extends Entity
+
+Let's define a base model class `BaseEntity` in **src/models/base-entity.ts**.
+It will be used an input to the mixin later.
+
+{% include code-caption.html content="src/models/base-entity.ts" %}
+
+```ts
+import {Entity} from '@loopback/repository';
```

Comment:
(I know it's a rare case and outside of the scope of this PR :p) If we apply the mixin to a `Model` instead of `Entity`, also need to define a BaseModel class to extend the abstract [`Model`](https://github.com/strongloop/loopback-next/blob/master/packages/repository/src/model.ts#L203) right?

Reply:
> (I know it's a rare case and outside of the scope of this PR :p) If we apply the mixin to a `Model` instead of `Entity`, also need to define a BaseModel class to extend the abstract [`Model`](https://github.com/strongloop/loopback-next/blob/master/packages/repository/src/model.ts#L203) right?

Out of scope for this PR. Most people will use Entity and EntityCrudRepository. This guide describes HOW to create a mixin in LB4.  With that knowledge, they can create whatever mixins they want, and targeting whatever model objects they want. :)